x/gov/keeper/proposal.go

```diff
@@ -76,6 +76,8 @@ func (keeper Keeper) SubmitProposal(ctx context.Context, messages []sdk.Msg, met
 		// ref: https://github.com/cosmos/cosmos-sdk/pull/10868#discussion_r784872842
 		if msg, ok := msg.(*v1.MsgExecLegacyContent); ok {
 			cacheCtx, _ := sdkCtx.CacheContext()
+			// This is a simulation context, so we should set the isCheckTx flag.
+			cacheCtx = cacheCtx.WithIsCheckTx(true)
```

Comment:
is this needed for something within wasm?

Reply:
yes @NoahSaso could you explain please what's the reason for this? As far as I can see, `IsCheckTX`  is only used for validator fees which are totally unrelated with the execution context of a gov msg...

Reply:
gentle ping @NoahSaso

Reply:
i've just been messing with the binary to explore state indexing and it seemed like `isCheckTx` is used for simulated transactions, so i figured this was missing here. i used that flag to determine whether or not to index a transaction, and it was giving some false positives when simulating wasm gov executions. not a big deal

Reply:
hmm i dont think this is needed, state streaming is the main way how state indexing is done in the sdk. I dont think we should add this unless we can have a test case of how its missing